packages/packages/core/frontend-handlers/src/handlers-registry.ts

```diff
@@ -1,4 +1,8 @@
-type Handler = ( params: { element: Element; signal: AbortSignal } ) => ( () => void ) | undefined;
+type Handler = ( params: {
+	element: Element;
+	signal: AbortSignal;
+	settings: Record< string, unknown >;
```

Comment:
Maybe 
```suggestion
	settings: TSettings;
```
 and allow user pass it as generic to the register function, `Record< string, unknown >` can be the default if nothing passed